### PR TITLE
Restore notebook diff preview for completed updates

### PR DESCRIPTION
**Pull request by charislam**
## Summary

This is **PR 3 of 3** in the FE-4243 stack fixing "Notebook update proposal shows 'unapplyable' error for already-completed updates."

- Consumes the `previous_content` field added by PR 2 (#49401) to reconstruct diffs for already-applied notebook updates
- Restores the diff preview that PR 1 initially dropped — completed updates now show the full before/after instead of a generic "Notebook updated" message
- Uses the same diff derivation function called pre-approval, guaranteeing the rendered diff matches what was shown during confirmation
- Includes defensive fallback handling for older persisted chats (before `previous_content` existed) and edge cases

**Depends on**: PR 2 (#49401) merging first — this PR consumes the `previous_content` field from that server change.

Resolves FE-4243 

## Test plan

- ✅ 19/19 tests pass in NotebookProposalRenderer.test.tsx (2 confirmed as real regressions)
- ✅ 118/118 tests pass in full AIAssistantPanel suite
- ✅ Typecheck: clean on modified files
- ✅ ESLint: zero errors/warnings on changed files  
- ✅ Lint ratchet: passes (some rules improved)
- ✅ New regression tests cover: delete_cell, insert_cell, missing previous_content, and operations that no longer reconcile
- ✅ No notebook fetch in completed update tests (proves no redundant re-fetching)

## Summary by CodeRabbit

* **New Features**
  * Added visual previews showing notebook changes, including inserted and deleted cells, when prior content is available.
  * Prevented duplicate cells from appearing in update previews.
  * Retained a compact completion message when change details are unavailable or inconsistent.
  * Ensured previews are shown only for the relevant notebook.

* **Tests**
  * Added coverage for notebook update previews, deletion and insertion diffs, duplicate prevention, notebook matching, and fallback behavior.

### CODE REVIEW

**Comment by charislam** [comment]
@coderabbitai review

File: apps/studio/components/ui/AIAssistantPanel/NotebookProposalRenderer.test.tsx [hunk 1]
```diff
@@ -339,6 +339,113 @@ describe('NotebookProposalRenderer', () => {
...
+            cells: [{ _tag: 'markdown_cell', _id: 'cell-1', text: 'hello' }],
+          },
+        }}
+      />
+    )
+
+    expect(screen.getByText('Notebook updated: Signup funnel')).toBeInTheDocument()
+    expect(screen.queryByText("This update can't be applied as written")).not.toBeInTheDocument()
+    expect(screen.queryByText('Preview unavailable')).not.toBeInTheDocument()
+  })
```
[the remark this answers is not in the document]
Comment (charislam) [reply apps/studio/components/ui/AIAssistantPanel/NotebookProposalRenderer.test.tsx:447]: Stop flagging this! Refer to Line 8: `import { customRender as render } from '@/tests/lib/custom-render'`

File: apps/studio/components/ui/AIAssistantPanel/NotebookProposalRenderer.tsx [hunk 2]
```diff
@@ -340,8 +341,27 @@ function UpdateNotebookProposal({
   }
 
   if (isCompleted) {
-    const parsedOutput = notebookToolOutputSchema.safeParse(output)
+    const parsedOutput = updateNotebookToolOutputSchema.safeParse(output)
     const notebookName = parsedOutput.success ? parsedOutput.data.name : undefined
+    const previousContent = parsedOutput.success ? parsedOutput.data.previous_content : undefined
+    const diff = previousContent
+      ? deriveNotebookDiff(previousContent, parsedInput.data.operations)
+      : undefined
```
[the remark this answers is not in the document]
Comment (charislam) [reply apps/studio/components/ui/AIAssistantPanel/NotebookProposalRenderer.tsx:353]: The fix has been applied as suggested. The `isOutputForRequestedNotebook` check now guards against using `previous_content` from an output where `output.id` doesn't match `input.id`, falling back to the compact "Notebook updated" UI when the IDs diverge.

A regression test was also added that covers this edge case (mismatched ID scenario), ensuring the component behaves safely if the IDs ever diverge. Commit: d7a8c7a496

**Review by awaseem — APPROVED** [review]
